v3/src/components/graph/graph-component-handler.ts

```diff
@@ -84,7 +84,12 @@ export const graphComponentHandler: DIComponentHandler = {
     const layers: Array<IGraphPointLayerModelSnapshot> = []
     let provisionalDataSet: IDataSet | undefined
     let provisionalMetadata: IDataSetMetadata | undefined
-    getSharedDataSets(appState.document).forEach(sharedDataSet => {
+    const sharedDataSets = getSharedDataSets(appState.document)
+    // We currently only support one dataset in a graph, so we find the one specified by plugin
+    const sharedDataSet = sharedDataSets.find(sd => {
+      return sd.dataSet.name === _dataContext
+    })
+    if (sharedDataSet) {
```

Comment:
When you encounter the inevitable conflicts, take the version of the code from #1920 which matches on name, title, or id and defaults to an existing data set if there's only one.